src/o365/spo/commands/search/search.ts

```diff
@@ -0,0 +1,434 @@
+      },
+      {
+        option:'--rankingModelId <rankingModelId>',
+        description: 'The ID of the ranking model to use for the query.'
+      },
+      {
+        option: '--startRow <startRow>',
+        description: 'The first row that is included in the search results that are returned. You use this parameter when you want to implement paging for search results.'
+      },
+      {
+        option: '--properties <properties>',
+        description: 'Additional properties for the query. GET requests support only string values. POST requests support values of any type.'
+      },
+      {
+        option: '--sourceName <sourceName>',
+        description: 'Specified the name of the result source to be used to run the query.'
+      },
+      {
+        option: '--refiners <refiners>',
+        description: 'The set of refiners to return in a search result.'
+      },
+      {
+        option: '--web <web>',
+        description: 'The web against which we want to execute the query. If the parameter is not defined, the query is executed against the web that\'s used when logging in to the SPO environment.'
+      },
+      {
+        option: '--hiddenConstraints <hiddenConstraints>',
+        description: 'The additional query terms to append to the query.'
+      },
+      {
+        option: '--clientType <clientType>',
+        description: 'The type of the client that issued the query.'
+      },
+      {
+        option: '--enablePhonetic',
+        description: 'A Boolean value that specifies whether the phonetic forms of the query terms are used to find matches. (Default = false).'
+      },
+      {
+        option: '--processBestBets',
+        description: 'A Boolean value that specifies whether to return best bet results for the query.'
+      },
+      {
+        option: '--enableQueryRules',
+        description: 'A Boolean value that specifies whether to enable query rules for the query. '
+      },
+      {
+        option: '--processPersonalFavorites',
+        description: 'A Boolean value that specifies whether to return personal favorites with the search results.'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.query) {
+        return 'Required parameter query missing';
+      }
+      if (args.options.sourceId && !Utils.isValidGuid(args.options.sourceId)) {
+        return `${args.options.sourceId} is not a valid GUID`;
+      }
+      if (args.options.rankingModelId && !Utils.isValidGuid(args.options.rankingModelId)) {
+        return `${args.options.rankingModelId} is not a valid GUID`;
+      }
+      if(args.options.sortList && !Utils.isRegExMatch(args.options.sortList,"^([a-z0-9_]+:(ascending|descending))(,([a-z0-9_]+:(ascending|descending)))*$")) {
```

Comment:
Since `isRegExMatch` is used only in one place and can be simplified to `/^regex$/.test()` I don't think there is much value in defining a separate method for it.